src/providers/Elsa.DistributedLocking.AzureBlob/AzureBlobLockProvider.cs

```diff
@@ -0,0 +1,199 @@
+                await blob.UploadTextAsync(string.Empty,cancellationToken).ConfigureAwait(false);
+
+            _renewTimer = new Timer(RenewLeases, null, _renewInterval, _renewInterval);
+            _logger.LogInformation($"Lock provider will try to acquire lock for {resourceName}");
+
+            if (_leaseSemaphore.WaitOne())
+            {
+                try
+                {
+                    var leaseId = await blob.AcquireLeaseAsync(_leaseTime, null, cancellationToken).ConfigureAwait(false);
+                    _lockedBlobs.Add(new LockedBlob { Blob = blob, LeaseId = leaseId, Identifier = resourceName });
+
+                    _logger.LogInformation($"Lock provider acquired lock for {resourceName}");
+                    return true;
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogWarning($"Failed to acquire lock for {resourceName}. Reason > {ex}");
+                    return false;
+                }
+                finally
+                {
+                    _leaseSemaphore.Set();
+                }
+            }
+            return false;
+        }
+
+        public Task ReleaseLockAsync(string name, CancellationToken cancellationToken = default)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            return ReleaseLeaseAsync(name, cancellationToken);
+        }
+        private async Task ReleaseLeaseAsync(string name, CancellationToken cancellationToken =default)
+        {
+            var resourceName = $"{Prefix}:{name}";
+            _logger.LogInformation($"Lock provider will try to release lock for {resourceName}");
+
+            if (_leaseSemaphore.WaitOne())
+            {
+                try
+                {
+                    var lockedBlob = _lockedBlobs.FirstOrDefault(x => x.Identifier == resourceName);
+
+                    if (lockedBlob != null)
+                    {
+                        try
+                        {
+                            await lockedBlob.Blob.ReleaseLeaseAsync(AccessCondition.GenerateLeaseCondition(lockedBlob.LeaseId),cancellationToken)
+                                            .ConfigureAwait(false);
+
+                            _logger.LogInformation($"Lock provider released lock for {resourceName}");
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogWarning($"Failed to release lock for {resourceName}. Reason > {ex}");
+                        }
+                        _lockedBlobs.Remove(lockedBlob);
+                    }
+                }
+                finally
+                {
+                    _leaseSemaphore.Set();
+                    _renewTimer?.Dispose();
+                }
+            }
+        }
+
+        private CloudBlobContainer CloudBlobContainer
```

Comment:
Any chance we can make this an async method so we can leverage the non-blocking methods of the blob container?

Reply:
Hmm, but dont think it will give you much benefit, because  -There are only two non-blocking methods `_cloudBlobContainer.Exists()` and `_cloudBlobContainer.CreateIfNotExists` and theoretically it is very rare that code execution will flow till that, so what I mean is it is not the HotPath.